packages/utils/telemetry-utils/src/logger.ts

```diff
@@ -249,6 +249,32 @@ export class ChildLogger extends TelemetryLogger {
         namespace?: string,
         properties?: ITelemetryProperties,
         propertyGetters?: ITelemetryPropertyGetters): TelemetryLogger {
+        // if we are creating a child of a child, rather than nest, which will increase
+        // the callstack overhead, just generate a new logger that includes everything from the previous
+        if (baseLogger instanceof ChildLogger) {
+            const combinedProperties =
+                baseLogger.properties === undefined && properties === undefined
+                    ? undefined
+                    : {
+                        ...baseLogger.properties,
+                        ... properties,
+                    };
+            const combinedGetters =
+                baseLogger.propertyGetters === undefined && propertyGetters === undefined
+                    ? undefined
+                    : {
+                        ...baseLogger.propertyGetters,
+                        ... propertyGetters,
+                    };
+
+            return new ChildLogger(
+                baseLogger.logger,
```

Comment:
Is there a good way to test that a grandchild logger's base logger is the grandparent? I see that baseLogger is protected, so there'd need to be some trickery.  Fine to punt, just checking.